Description: Skip any warnings from QEMU while waiting for UEFI output
Author: dann frazier <dannf@ubuntu.com>
Last-Update: 2019-08-01

Index: edk2-0~20190606.20d2e5a1/qemu-ovmf-secureboot-1-1-3/ovmf-vars-generator
===================================================================
--- edk2-0~20190606.20d2e5a1.orig/qemu-ovmf-secureboot-1-1-3/ovmf-vars-generator
+++ edk2-0~20190606.20d2e5a1/qemu-ovmf-secureboot-1-1-3/ovmf-vars-generator
@@ -108,6 +108,10 @@ def enroll_keys(args):
     read = p.stdout.readline()
     if b'char device redirected' in read:
         read = p.stdout.readline()
+    # Skip passed QEMU warnings, like the following one we see in Ubuntu:
+    # qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.vmx [bit 5]
+    while b'qemu-system-x86_64: warning:' in read:
+        read = p.stdout.readline()
     if args.print_output:
         print(strip_special(read), end='')
         print()
